features/support/commits_finder.rb

```diff
@@ -0,0 +1,49 @@
+# Encapsulates finding Git commits in flexible ways
+#
+# After initializing with the desired attributes for the commit list,
+# run as many #find_* methods as you want to build up your commit list.
+# When done, call #to_a to get the results as a Cucumber compatible array.
+class CommitsFinder
```

Comment:
I still don't like this architecture. 
A finder to me shouldn't have internal state, a `find_*` method should return whatever commits it finds.
That return value could be an array, a hash, or some other data structure like CommitList (which can have a `to_table` function which returns the commits as a cucumber table).